source/module_hamilt_pw/hamilt_pwdft/forces.cpp

```diff
@@ -920,7 +920,7 @@ void Forces<FPTYPE, Device>::cal_force_nl(ModuleBase::matrix& forcenl,
     // ModuleBase::ComplexMatrix vkb1(nkb, this->npwx);
     resmem_complex_op()(this->ctx, vkb1, this->npwx * nkb, "Force::vkb1");
     // init additional params
-    FPTYPE *force = nullptr, *d_wg = nullptr, *deeq = nullptr, *gcar = nullptr;
+    FPTYPE *force = nullptr, *d_wg = nullptr, *gcar = nullptr, *deeq = GlobalC::ppcell.get_deeq_data<FPTYPE>();
```

Comment:
Maybe you can split definitions of variables to multi lines, it is advice from @mohanchen .